src/main/java/bio/terra/service/filedata/FileMetadataUtils.java

```diff
@@ -77,10 +80,33 @@ public FireStoreDirectoryEntry makeDirectoryEntry(String lookupDirPath) {
   public String makeLookupPath(String fullPath) {
     String temp = StringUtils.prependIfMissing(fullPath, "/");
     temp = StringUtils.removeEnd(temp, "/");
-    return ROOT_DIR_NAME + temp;
+    temp = StringUtils.prependIfMissing(temp, ROOT_DIR_NAME);
+    return temp;
   }
 
   public String makePathFromLookupPath(String lookupPath) {
     return StringUtils.removeStart(lookupPath, ROOT_DIR_NAME);
   }
+
+  public List<String> findNewDirectoryPaths(
+      List<FireStoreDirectoryEntry> datasetEntries, LRUMap<String, Boolean> pathMap) {
```

Comment:
I don't see a need to use the more specific type `LRUMap` here. This code doesn't need to care what type of `Map` the caller passes in.
```suggestion
      List<FireStoreDirectoryEntry> datasetEntries, Map<String, Boolean> pathMap) {
```

Reply:
This is also now covered in https://broadworkbench.atlassian.net/browse/DR-2145